test/gpu/jit.cpp

```diff
@@ -356,4 +356,50 @@ TEST_CASE(compile_math)
     });
 }
 
+// NOLINTNEXTLINE
+const std::string assert_template = R"__migraphx__(
+#include <migraphx/kernels/math.hpp>
+#include <migraphx/kernels/types.hpp>
+using namespace migraphx;
+extern "C" {
+__global__ void kernel(void*) 
+{
+    static_assert(numeric_max<${type}>() == std::numeric_limits<${type}>::max(), "");
+    static_assert(numeric_lowest<${type}>() == std::numeric_limits<${type}>::lowest(), "");
```

Comment:
Ha good point... Otherwise why not directly call `numeric_limits<>()`.

> {"max", as.max()}, {"lowest", as.min()}

That was my first attempt yesterday. But turns out it doesn't know how to convert `as.max()` or `as.min()` into a string. So what's the utility that would convert a arbitrary type into a string correctly?

Reply:
Done in dc852557c2ab6cb6c44e4ac140225c1413906a26